packages/datadog-plugin-aws-sdk/src/services/s3.js

```diff
@@ -18,6 +25,39 @@ class S3 extends BaseAwsSdkPlugin {
       bucketname: params.Bucket
     })
   }
+
+  addSpanPointers (span, response) {
+    const request = response?.request
+    const operationName = request?.operation
+    if (!['putObject', 'copyObject', 'completeMultipartUpload'].includes(operationName)) {
+      // We don't create span links for other S3 operations.
+      return
+    }
+
+    // AWS v2: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html
+    // AWS v3: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/
+    const bucketName = request?.params?.Bucket
+    const objectKey = request?.params?.Key
+    const eTag =
+      response?.ETag || // v3 PutObject & CompleteMultipartUpload
+      response?.CopyObjectResult?.ETag || // v3 CopyObject
+      response?.data?.ETag || // v2 PutObject & CompleteMultipartUpload
+      response?.data?.CopyObjectResult?.ETag // v2 CopyObject
+
+    if (!bucketName || !objectKey || !eTag) {
+      log.debug('Unable to calculate span pointer hash because of missing parameters.')
+      return
+    }
+
+    const pointerHash = generateS3PointerHash(bucketName, objectKey, eTag)
+    const attributes = {
+      'ptr.kind': S3_PTR_KIND,
+      'ptr.dir': SPAN_POINTER_DIRECTION.DOWNSTREAM,
+      'ptr.hash': pointerHash,
+      'link.kind': SPAN_LINK_KIND
```

Comment:
now that i'm back here, it's a bit weird that we're telling the span pointer (through `addSpanPointer`) that it's kind is `SPAN_LINK_KIND`